askama_shared/src/parser.rs

```diff
@@ -128,6 +128,37 @@ pub struct CondTest<'a> {
     pub expr: Expr<'a>,
 }
 
+/// Skips over the any amout of `inner` until `end` was found.
+/// Returns a tuple of the skipped string and the found end marker.
+fn skip_till<'a, I, O>(
+    mut inner: impl FnMut(&'a [u8]) -> IResult<&'a [u8], I>,
+    end: impl FnMut(&'a [u8]) -> IResult<&'a [u8], O>,
+) -> impl FnMut(&'a [u8]) -> IResult<&'a [u8], (&'a [u8], O)> {
+    let mut opt_end = opt(end);
+    move |start: &'a [u8]| {
+        let mut i = start;
+        loop {
+            if let (j, Some(o)) = opt_end(i)? {
+                let skipped = &start[..start.offset(i)];
+                return Ok((j, (skipped, o)));
+            } else {
+                i = inner(i)?.0;
+            }
+        }
+    }
+}
+
+/// Succeeds if parser matches, but the input is not consumed.
```

Comment:
Is this different from https://docs.rs/nom/7.1.0/nom/combinator/fn.peek.html?

Reply:
Huh, I swear I looked for this function but I didn't find it.